File: hard/license/license.java
package ejercicios;

import java.util.Collections;
import java.util.LinkedList;
import java.util.Scanner;

/**
 *07/05/2021
 * @author Andre Peralta
 */
public class metodo {
    public static void main(String[] args) { 

        Scanner input = new Scanner(System.in);
        
        System.out.println("Mi name : ");
        String mi_name = input.next();
        
        System.out.println("Number of agents : ");
        int agents = input.nextInt();
        
        input.nextLine();
        
        System.out.println("Names : ");
        String people_names = input.nextLine();

        //Separamos el string por espacios
        String[] p_names = people_names.split("\\s+");
        
        LinkedList<String> names_list = new LinkedList();
        
        //Pasamos de String[] a LinkedList para usar el metodo sort
        for(String name: p_names){
            names_list.add(name.toLowerCase());
            //En minusculas para que haya problemas co sort
        }
        names_list.add(mi_name.toLowerCase());
        Collections.sort(names_list);
        
        //Obtenemos la posición de mi nombre con la lista ya ordenada
        int posicion = names_list.indexOf(mi_name);
        //Hacemos el pequeno calculo con respecto al número de agentes
        double calculation = Math.ceil(posicion+1/agents)*20;
        
        System.out.println((int)calculation);

   }
}

File: hard/license/license.kt
fun minutes(mi_name:String,names:String,agents:Int):Double{
    
    //Creamos una lista a partir del string
    var names_list = names.split(" ")
    //Agregamos mi nombre a la lista
    names_list += mi_name
    //Ordenamos la lista
    names_list = names_list.sorted()
    //Obtenemos la posición de mi nombre con la lista ya ordenada
    var posicion = names_list.indexOf(mi_name)
    //Hacemos el pequeno calculo con respecto al número de agentes
    var var1:Double = (posicion+1)/agents.toDouble()//2/5 es 2 a menos que use esto
    var time = (var1*20)+(var1*20)%20

    return(time)
}

fun main(args: Array<String>){
	
    println("Mi name : ")
    var mi_name = readLine()
    
    println("Names list : ")
    var names = readLine()
    
    println("Agents : ")
    var agents = readLine()!!.toInt()	
    
    println(minutes(mi_name,names,agents))    

    
    

}
